Essentials/src/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -338,6 +347,13 @@ public void addFireworkMeta(final CommandSource sender, final boolean allowShort
         }
     }
 
+    private Color getColorFromRgbHexString(String color) {
+        if (hexStringPattern.matcher(color).find()) {
+            return Color.fromRGB(Integer.decode(color.substring(color.indexOf("[") + 4, color.indexOf("]") - 1)));
```

Comment:
Use groups to extract the color portion of the string.

Reply:
What will happen if the value passed to `Integer.decode()` is not a valid hex integer? Will it through an exception or return `null`?